src/ChainRouterBcLayer.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Routing;
+
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+$refl = new \ReflectionClass(UrlGeneratorInterface::class);
+$generateMethod = $refl->getMethod('generate');
+$methodParameters = $generateMethod->getParameters();
```

Comment:
there are all global variables, which has me worried a bit. for that reason, i would prefer to rewrite this as

```
$urlGeneratorNameParameter = array_shift((new \ReflectionClass(UrlGeneratorInterface::class)->getMethod('generate')->getParameters());
```

Reply:
I will need 1 more variable here as php will throw a notice "Only variables should be passed by reference". This will also make the tests fail